galasa-ui/src/utils/3270/get3270Screenshots.ts

```diff
@@ -65,24 +65,45 @@ export function populateFlattenedZos3270TerminalDataAndAllImageData(images: any[
     const result = splitScreenAndTerminal(id);
 
     flattenedZos3270TerminalData.push({
-      id: image.id,
+      id: id,
       Terminal: result.terminalName,
       screenNumber: result.screenNumber,
     });
 
     // Populate all image data for screenshot rendering.
     const imageFields: TerminalImageField[] = image.fields
       // Filter out any image fields that are missing a row or column.
-      .filter((imageField: any) => imageField.row != null && imageField.column != null)
-      .map((imageField: any) => ({
+      .filter(
+        (imageField: TerminalImageField) => imageField.row != null && imageField.column != null
+      )
+      .map((imageField: TerminalImageField) => ({
         row: imageField.row,
         column: imageField.column,
-        text: imageField.contents[0]?.text ?? '',
+        contents: imageField.contents,
+        unformatted: imageField.unformatted ? imageField.unformatted : undefined,
+        fieldProtected: imageField.fieldProtected ? imageField.fieldProtected : undefined,
+        fieldNumeric: imageField.fieldNumeric ? imageField.fieldNumeric : undefined,
+        fieldDisplay: imageField.fieldDisplay ? imageField.fieldDisplay : undefined,
+        fieldIntenseDisplay: imageField.fieldIntenseDisplay
+          ? imageField.fieldIntenseDisplay
+          : undefined,
+        fieldSelectorPen: imageField.fieldSelectorPen ? imageField.fieldSelectorPen : undefined,
+        fieldModified: imageField.fieldModified ? imageField.fieldModified : undefined,
+        foregroundColor: imageField.foregroundColor ? imageField.foregroundColor : undefined,
+        backgroundColor: imageField.backgroundColor ? imageField.backgroundColor : undefined,
+        highlight: imageField.highlight ? imageField.highlight : undefined,
       }));
 
     allImageData.push({
       id: image.id,
-      imageFields: imageFields,
+      sequence: image.sequence ? image.sequence : undefined,
+      inbound: image.inbound ? image.inbound : undefined,
+      type: image.type ? image.type : undefined,
+      imageSize: image.imageSize,
+      cursorRow: image.cursorRow ? image.cursorRow : undefined,
+      cursorColumn: image.cursorColumn ? image.cursorColumn : undefined,
+      aid: image.aid ? image.aid : undefined,
+      fields: imageFields,
```

Comment:
Minor - If the `TerminalImageField` properties are defined to allow undefined values like:

```ts
fieldProtected?: boolean
```

then I think this can be made more readable by just setting the value from the `imageField` object, like:

```ts
fieldModified: imageField.fieldModified
```

rather than the explicit ternary operations:

```ts
fieldModified: imageField.fieldModified ? imageField.fieldModified : undefined
```